File: onlinebookstore/app/models/book_model.py
from app.extensions import db

class Book(db.Model):
    __tablename__ = 'books'
    id = db.Column(db.Integer, primary_key=True)
    title = db.Column(db.String(100), nullable=False)
    author = db.Column(db.String(100), nullable=False)
    genre = db.Column(db.String(50))
    price = db.Column(db.Float, nullable=False)
    publication_date = db.Column(db.Date)
    bookstore_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)

    orders = db.relationship('Order', backref='book', lazy=True)
    reviews = db.relationship('Review', backref='book', lazy=True)

File: onlinebookstorefront/src/app/models/book.model.ts
export interface Book {
    id: number;
    title: string;
    author: string;
    genre?: string;  // Optional as it might not always be provided
    price: number;
    publication_date?: string;  // Using string to match a date format (you can parse it later)
    bookstore_id: number;
    orders?: any[];  // Optional, could represent the orders related to this book
    reviews?: any[];  // Optional, could represent the reviews related to this book
  }
  